App.js

```diff
@@ -204,26 +204,7 @@ function ChatScreen({route, navigation}) {
           <Text style={{color: 'blue', fontSize: 18, marginBottom: 10}}>Link to another chat</Text>
         </Pressable>
         <Pressable
-          onPress={() => {
-            // This is fairly complex but seems to be the way we can "push" a new stack on and add the Settings screen in the routes
-            navigation.dispatch((state) => {
-              const newState = {...state};
-              const routes = [...newState.routes];
-              routes.push({
-                name: 'SettingsStack',
-                state: {
-                  index: 1,
-                  routes: [{name: 'Settings'}, {name: 'About'}]
-                }
-              });
-              const newestState = {
-                ...newState,
-                routes,
-                index: routes.length - 1,
-              };
-              return CommonActions.reset(newestState);
-            });
-          }}
+          onPress={() => navigation.push('SettingsStack', { screen: 'About', initial: false })}
```

Comment:
@marcaaron Oh interesting, I havent run into this, because at least what I tried you need to do these steps after you load the page with navigation state cleared/ fresh new. If you get some other routes to the states, then this go back functionality will work as expected.

I am not sure why this is happening, will have to look more. cc @adamgrzybowski if you would have some time to check this out, maybe this has obvious solution.

Reply:
This is an interesting one because the state seems to be updated correctly so not sure why this is behaving oddly. Looking into it more.

Reply:
It looks like it counts one action forward so it can go only one back. It behaves differently if the browser tab isn't "fresh". I am not sure how to check the history in the react-native-web yet.

Reply:
Ok, we may have to push a state to the browser history to match with the react-navigation state. It should be possible https://developer.mozilla.org/en-US/docs/Web/API/History_API?redirectlocale=en-US&redirectslug=DOM%2FManipulating_the_browser_history#Adding_and_modifying_history_entries

Reply:
Yeah, that seems to be the case here. So the main issue here is `pop` method in the `About` page seems to also call `back()` for the history object so we can go back only one time.
I looked into it and found a solution that matches what Vit had already implemented.

In the ChatScreen we can do `navigation.push('SettingsStack', { screen: 'About'})` and in the About screen we can use the replace method replacing the `About` screen with the `SettingsStack` screen when `Up` action is taken, which by default renders the `Settings` screen. The reason we are replacing it with the `SettingsStack` screen and not the `Settings` screen is otherwise we will introduce the bug describer [here](https://github.com/marcaaron/navigation-test/pull/5#discussion_r1044689959).

Reply:
@adamgrzybowski 
> Ok, we may have to push a state to the browser history to match with the react-navigation state. It should be possible https://developer.mozilla.org/en-US/docs/Web/API/History_API?redirectlocale=en-US&redirectslug=DOM%2FManipulating_the_browser_history#Adding_and_modifying_history_entries

Ohh yes, good find. We will have to manually sync the navigation state and browser history if we can't find a simple solution.

Reply:
Oh, so React Navigation (on the web) uses a memory history here:

https://github.com/react-navigation/react-navigation/blob/main/packages/native/src/createMemoryHistory.tsx

I think this is the main place where most web history stuff happens. Taken together with `useLinking` here:

https://github.com/react-navigation/react-navigation/blob/main/packages/native/src/useLinking.tsx

This will put the whole "web history" thing together for anyone (well, it's a but complicated, but that's where event listeners are set things are pushed, replaced, etc).